src/wp-includes/l10n.php

```diff
@@ -132,30 +132,25 @@ function determine_locale() {
 	 */
 	$determined_locale = apply_filters( 'pre_determine_locale', null );
 
-	if ( ! empty( $determined_locale ) && is_string( $determined_locale ) ) {
+	if ( $determined_locale && is_string( $determined_locale ) ) {
 		return $determined_locale;
 	}
 
-	$determined_locale = get_locale();
-
-	if ( is_admin() ) {
-		$determined_locale = get_user_locale();
-	}
-
-	if ( isset( $_GET['_locale'] ) && 'user' === $_GET['_locale'] && wp_is_json_request() ) {
+	if (
+		( isset( $_GET['_locale'] ) && 'user' === $_GET['_locale'] && wp_is_json_request() ) ||
+		is_admin()
+	) {
 		$determined_locale = get_user_locale();
+	} else {
+		$determined_locale = get_locale();
```

Comment:
In my PR, I don't do this. What we are trying to avoid here is, running `get_locale` and then it gets overridden if `pagenow` is set. 

Moving `$determined_locale = get_locale();` to after line 155 saves that.

Reply:
In your PHP an `is_null` check is needed, which is also not great.

I think 6397d94436d33e20ee0078f566e981b95fbbffb6 solves this better.